### DISCUSSION TITLE
Postgres hook uses deprecated parameter from base_aws.py

### DISCUSSION THREAD

**Discussion by akennedy4155** [apache/airflow#21845] [General]
### Apache Airflow Provider(s)

postgres

### Versions of Apache Airflow Providers

Most recent

### Apache Airflow version

2.2.3 (latest released)

### Operating System

macOS Big Sur 11.3.1

### Deployment

Astronomer

### Deployment details

Nothing else needed here.  The source means that this is deployment agnostic I believe.

### What happened

Postgres hook calls AwsBaseHook, and uses the client_type parameter from the conn cached property.  Using client_type as a parameter here is deprecated, and it should be specified as part of the class initialization I believe.

Warning message is: ```client_type is deprecated. Set client_type from class attribute.```

### What you expected to happen

No deprecated calls or warning messages.

### How to reproduce

Use postgres_hook.get_iam_token

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by eladkal** [comment]
I don't see the issue. The reference to `client_type` is OK
https://github.com/apache/airflow/blob/3b4c26eb3a1c8d4938be80ab7fa0711561e91f8f/airflow/providers/postgres/hooks/postgres.py#L181

Can you provide example DAG to reproduce the warning?

**Comment by potiuk** [comment]
Converted it to a discussion until more information is provided. Please epxlain how to reproduce it @akennedy4155
